File: app/src/main/java/de/gaffga/android/fragments/MaxHeightRecyclerView.java
package de.gaffga.android.fragments;

import android.content.Context;
import android.util.AttributeSet;
import androidx.recyclerview.widget.RecyclerView;

public class MaxHeightRecyclerView extends RecyclerView {

    private int maxHeightPx = Integer.MAX_VALUE;

    public MaxHeightRecyclerView(Context context) {
        super(context);
    }

    public MaxHeightRecyclerView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MaxHeightRecyclerView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public void setMaxHeight(int maxHeightPx) {
        this.maxHeightPx = maxHeightPx;
        requestLayout();
    }

    public int getMaxHeight() {
        return maxHeightPx;
    }

    @Override
    protected void onMeasure(int widthSpec, int heightSpec) {
        super.onMeasure(widthSpec, heightSpec);
        if (getMeasuredHeight() > maxHeightPx) {
            setMeasuredDimension(getMeasuredWidth(), maxHeightPx);
        }
    }
}

File: app/src/main/kotlin/at/priv/graf/zazentimer/fragments/MaxHeightRecyclerView.kt
package at.priv.graf.zazentimer.fragments

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.RecyclerView

class MaxHeightRecyclerView : RecyclerView {

    private var maxHeightPx: Int = Integer.MAX_VALUE

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyle: Int) : super(context, attrs, defStyle)

    fun setMaxHeight(maxHeightPx: Int) {
        this.maxHeightPx = maxHeightPx
        requestLayout()
    }

    fun getMaxHeight(): Int = maxHeightPx

    override fun onMeasure(widthSpec: Int, heightSpec: Int) {
        super.onMeasure(widthSpec, heightSpec)
        if (getMeasuredHeight() > maxHeightPx) {
            setMeasuredDimension(getMeasuredWidth(), maxHeightPx)
        }
    }
}
